### DISCUSSION TITLE
npm dependency updates not getting recognized - misconfiguration?

### DISCUSSION THREAD

**Discussion by DManstrator** [renovatebot/renovate#12671] [Request Help] [answered]
Hello.

I currently face the issue that no Merge Requests are created for npm updates despite I enabled them explicitly.

Here the relevant part of the `renovate.config`:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "enabledManagers": [
    "npm",
    "maven"
  ],
  [...]
  "packageRules": [
    [...]
    {
      "matchManagers": ["npm"],
      "automerge": false
    }
  ]
}
```

**Perhaps important information:** When I initially created the `renovate.json`, only `maven` was in the `enabledManagers`. The Renovate Bot also created a first initial Merge Request where it said it detected the Maven package file. Later on, I added a `package.json` and added the `npm` entries to the `renovate.json`.

My `package.json` (for testing purposes) where each depenency is not on the latest version (missing a patch, minor and major version respectively):
```json
{
  "name": "test",
  "version": "1.0.0",
  "dependencies": {
    "randomstring": "1.2.0",
    "lodash": "4.16.6",
    "platform-detect": "2.0.3",
  }
}
```

I committed and pushed that files on last Friday, no updates so far.

Do you know what could be the issue? Do you need more information?

**Comment by rarkins** [comment]
If you're using the hosted app, please log into app.renovatebot.com and look at the logs. Pay attention to the "packageFiles with updates" log message which prints all detected package managers and dependencies.

**Reply by DManstrator** [reply]
Hello Rhys, thanks for the quick answer.

We are using the self-hosted option and the only logs I've found are those:

![image](https://user-images.githubusercontent.com/7572614/141986261-834137fa-078e-409d-a215-f1bb04328c65.png)

As said, I added `npm` to the `enabledManagers` after initializing the Renovate Bot. Could it be that this attribute was somehow cached or not updated?

Or is it an issue with my `package.json`? I recognized there was a mistake with it (there was a comma after the last dependency, see code above). Does the Renovate Bot check if the manager files are syntactically correct?

I also removed the `enabledManagers` section for testing purposes and after those two changes, Merge Requests finally appeared. So right now I can't say if it was infact an issue with the `package.json` or not. I will test that but the Renovate Bot has its next run configured in about 5 hours.

**Reply by viceice** [reply]
You need to run with `LOG_LEVEL=debug` env or check if the config writes a log file

**Reply by DManstrator** [reply]
I tried to replicate that with debug logs this time. I added the `enabledManagers` back and added a new dependency to test it. But this time, the Merge Request was created.

As said before, my `package.json` was invalid since I had a trailing comma on the last dependency. I add another dependency to the bottom and put a comma after it so the file is invalid again. And this time, no Merge Request was created. Even more funny, the previously opened Merge Request was closed by the bot (it also added ` - autoclosed` to the title.

**So it really seems that the invalid `package.json` was the reason, why the Renovate Bot created no Merge Requests.**

I pushed again without the trailing comma and will check if the Merge Request will be created now (and if the autoclosed MR will be reopened again). But as of now, I am sure that this was the problem.

Thank you both for the quick response though!

**Edit:**

Yes, exactly what I thought. After I fixed the `package.json`, the Merge Requests were opened again (for the autoclosed one a new one was created instead of the old one getting reopened again). So it really looks like the Renovate Bot checks if the dependency manager files are valid, at least for `npm`.

Still thanks for your quick help. :)

**Comment by DManstrator** [comment, marked as answer]
It was actually not an issue with my `renovate.json` but with the `package.json`.

As you can see in my initial question, there was a comma at the last dependency which made the file invalid.

>       "lodash": "4.16.6",
>       "platform-detect": "2.0.3",
>     }

After fixing that, Merge Requests for dependency updates for `npm` were created. As soon as the `package.json` got invalid again, it closed opened Merge Requests for `npm` too.

Thanks for your quick help though, much appreciated!
